vlingo-reactive-messaging-patterns/src/test/java/io/vlingo/reactive/messaging/patterns/competingconsumer/CompetingConsumerTest.java

```diff
@@ -0,0 +1,36 @@
+// Copyright © 2012-2018 Vaughn Vernon. All rights reserved.
+//
+// This Source Code Form is subject to the terms of the
+// Mozilla Public License, v. 2.0. If a copy of the MPL
+// was not distributed with this file, You can obtain
+// one at https://mozilla.org/MPL/2.0/.
+package io.vlingo.reactive.messaging.patterns.competingconsumer;
+/**
+ * CompetingConsumerTest provides an example of implementing
+ * and using a vlingo Router.
+ */
+import org.junit.Test;
+
+import io.vlingo.actors.Definition;
+import io.vlingo.actors.World;
+import io.vlingo.actors.testkit.TestUntil;
+
+public class CompetingConsumerTest {
+
+  @Test
+  public void testThatConsumersCompete() {
+    final World world = World.startWithDefaults("competing-consumer-test");
```

Comment:
@DaveMuirhead If you use `TestWorld` all the actors you create will have `TestMailbox` which is a blocking synchronous delivery (encapsulated method invocation). You can also use the `public TestState viewTestState()` on the actor to immediately read state. I don't like this, but I also like it. In a `TestWorld` this basically like POJO state test, but using a `Map` to get parts you need to see (review the `TestkitTest::testTesterWorldPing()`). This is not good because it enables querying state sans messaging, which is breaking the actor model, but if only used with `TestWorld` it's a good thing. Maybe that's the key--`Actor` must assert that `stage().world().isTestWorld()` is `true` or it throws. Also if you want to stay async you can use `Completes::await()`, which also has a timeout override.

This really needs to be added to the `docs.vlingo.io` on gitbook. I am going to invite you and make you an author :)